utils/sampler/lib/helper_widgets.dart

```diff
@@ -0,0 +1,354 @@
+                  ),
+                ),
+                Padding(
+                  padding: const EdgeInsetsDirectional.only(start: 8.0, end: 8.0),
+                  child: IconButton(
+                    tooltip: 'Copy path to clipboard',
+                    icon: const Icon(Icons.copy),
+                    onPressed: () {
+                      Clipboard.setData(ClipboardData(text: path));
+                    },
+                  ),
+                ),
+              ],
+            ),
+            Row(
+              mainAxisSize: MainAxisSize.min,
+              crossAxisAlignment: CrossAxisAlignment.center,
+              children: <Widget>[
+                Padding(
+                  padding: const EdgeInsetsDirectional.only(start: 8.0, end: 8.0),
+                  child: TextButton(
+                    child: Text('OPEN IN $_fileBrowserName'),
+                    onPressed: () {
+                      openFileBrowser(file ?? location);
+                    },
+                  ),
+                ),
+                for (final IdeType type in IdeType.values)
+                  Padding(
+                    padding: const EdgeInsetsDirectional.only(start: 8.0, end: 8.0),
+                    child: TextButton(
+                      child: Text('OPEN IN ${getIdeName(type).toUpperCase()}'),
+                      onPressed: () {
+                        openInIde(type, location, file: file, startLine: startLine);
+                      },
+                    ),
+                  ),
+              ],
+            ),
+          ],
+        ),
+      ),
+    );
+  }
+}
+
+/// A widget that shows a list of actions in a row, with a progress indicator
+/// when an action is in progress.
+class ActionPanel extends StatelessWidget {
+  const ActionPanel({
+    Key? key,
+    required this.children,
+    this.isBusy = false,
+  }) : super(key: key);
+
+  final List<Widget> children;
+  final bool isBusy;
+
+  @override
+  Widget build(BuildContext context) {
+    return Stack(
+      alignment: Alignment.center,
+      children: <Widget>[
+        Container(
+          height: 100,
+          margin: const EdgeInsets.all(8.0),
+          padding: const EdgeInsets.all(8.0),
+          decoration: const ShapeDecoration(
+            color: Colors.black12,
+            shape: RoundedRectangleBorder(
+              borderRadius: BorderRadius.all(
+                Radius.circular(10.0),
+              ),
+            ),
+          ),
+          child: Row(
+            mainAxisAlignment:
+                children.length < 2 ? MainAxisAlignment.center : MainAxisAlignment.spaceBetween,
+            crossAxisAlignment: CrossAxisAlignment.center,
+            children: children,
+          ),
+        ),
+        if (isBusy) const CircularProgressIndicator.adaptive(value: null),
+      ],
+    );
+  }
+}
+
+// The default Material-style Autocomplete text field.
+class AutocompleteField extends StatelessWidget {
```

Comment:
Wonder if this can use the new autocomplete widgets that have recently been added?

Reply:
Sure could! I thought they just hadn't been checked in yet. What are they called?

Reply:
Oddly enough `Autocomplete`: https://api.flutter.dev/flutter/material/Autocomplete-class.html

Reply:
Actually, turns out I am already using it.  The `AutocompleteField` is just a wrapper I made to wrap the text field that goes into the autocomplete widget.

Look at line 408 in this file, and you'll see where the `Autocomplete<File>` is.

Reply:
Ah, I didn't look closely enough. Never mind then 😄.